twine/settings.py

```diff
@@ -148,6 +150,18 @@ def password(self) -> Optional[str]:
         # Workaround for https://github.com/python/mypy/issues/5858
         return cast(Optional[str], self.auth.password)
 
+    @property
+    def verbose(self) -> bool:
+        return self._verbose
+
+    @verbose.setter
+    def verbose(self, verbose: bool) -> None:
+        """Initialize a logger based on the --verbose option."""
+        self._verbose = verbose
+        root_logger = logging.getLogger("twine")
+        root_logger.addHandler(logging.StreamHandler(sys.stdout))
+        root_logger.setLevel(logging.INFO if verbose else logging.WARNING)
```

Comment:
Seeing the switch from `capsys` to `caplog` in `test_upload.py` made me realize that we weren't actually testing that verbose messages were printed to `stdout`. Furthermore, the tests were still setting `Settings.verbose` directly, which wouldn't actually set up logging.

https://github.com/pypa/twine/blob/27f029bcb85969e5e7841958c81e5a476c514043/tests/test_upload.py#L71-L74

To keep the existing API and avoid rewriting all of the upload tests, I made `Settings.verbose` a `@property` and moved the logging setup there.